### DISCUSSION TITLE
Change listening port?

### DISCUSSION THREAD

**Discussion by joyfuI** [ZoeyVid/NPMplus#1197] [Q&A] [answered]
Hello.

Is there a way to change the HTTP and HTTPS ports?

I'm using a Synology NAS, and Synology already preempts ports 80 and 443, so I can't use them in "network_mode: host" unless I modify the system files.

**Comment by gadget78** [comment]
the UI port is within the docker.yml / or environment settings :) 
but you cannot change "port 80" as thats the main reason for NPMplus...

you could change it via the docker listener port (80:8080)

**Comment by Zoey2936** [comment, marked as answer]
there are some (complicated) ways to stop synologys nginx server to listen on port 80/443, I would recommend you to try that. Or as said above you can use network_mode bridge (please see the downsides at the top of the readme and #691) and change the ports through docker bindings.

Why can the ports in host mode not be changed?
- CAs only work with port 80/443, so only dns challange would be possible
- There was no reason yet, so I didn't wrote an option to change this. 
- but you can disable listing on port 80, so only 443 is used, other ports like NPMplus dashboard/goaccess can be changed

**Reply by joyfuI** [reply]
Thank you.

I thought I couldn't use "network_mode: bridge" because it says in the readme `since this fork has dependency on network_mode: host, please don't forget to open port 80/tcp, 443/tcp and 443/udp (and maybe 81/tcp) in your firewall`.

Since I'm using a router, I'm going to change the listening port to another port and forward it back to 80/443 via the port forwarding function.
